### PR TITLE
WebGLMultisampleRenderTarget: Make gl.invalidateFramebuffer() optional.

### PR DESCRIPTION
**Pull request by cabanier**
Related issue: #22967

**Description**

Some WebGL implementations (Edge and Safari on M1) don't handle invalidateFramebuffer correctly.
This has been addressed recently in Angle but broken implementations are still in the field.

This change allows the author to skip the broken logic. On mobile it will introduce a performance slowdown because it will flush multisampled pixels to main memory even though they will never be used.

This contribution is funded by [Oculus](https://oculus.com).

### CODE REVIEW

**Review by Mugen87 — APPROVED** [review]

**Comment by cabanier** [comment]
@Mugen87 Is there a way to be notified in 3-4 months to remove this workaround?

**Comment by Mugen87** [comment]
Maybe we just file an issue as a reminder.

**Comment by nkallen** [comment]
Hey -- I think we need to update the @types for this too

File: src/renderers/webgl/WebGLTextures.js [hunk 1]
```diff
@@ -1594,7 +1594,12 @@ function WebGLTextures( _gl, extensions, state, properties, capabilities, utils,
 				}
 
 				_gl.blitFramebuffer( 0, 0, width, height, 0, 0, width, height, mask, _gl.NEAREST );
-				_gl.invalidateFramebuffer( _gl.READ_FRAMEBUFFER, invalidationArray );
+
+				if ( ! renderTarget.skipInvalidateFramebuffer ) {
+
+					_gl.invalidateFramebuffer( _gl.READ_FRAMEBUFFER, invalidationArray );
+
+				}
```
Comment (mrdoob) [inline src/renderers/webgl/WebGLTextures.js:1602]: Can we check if the UA is Oculus Browser at init time and do this instead?

```js
if ( isOculusBrowser ) {

	_gl.invalidateFramebuffer( _gl.READ_FRAMEBUFFER, invalidationArray );

}
```

Comment (cabanier) [reply src/renderers/webgl/WebGLTextures.js:1602]: That will just make it faster for Oculus browser and not other mobile GPUs like Hololens.

Comment (mrdoob) [reply src/renderers/webgl/WebGLTextures.js:1602]: Ah, right...

Well, what I'm proposing is to make this automatic without the user having to know about it.

You said:

> Some WebGL implementations (Edge and Safari on M1) don't handle invalidateFramebuffer correctly.

But https://github.com/mrdoob/three.js/issues/22967#issuecomment-1018315477 says Chrome on Android is affected too.

Is there any way of making a UA check and have this code for 3-4 months?

**Pull request closed without merging**